pkg/resources/view.go

```diff
@@ -52,172 +59,550 @@ var viewSchema = map[string]*schema.Schema{
 		RequiredWith: []string{"or_replace"},
 	},
 	"is_secure": {
-		Type:        schema.TypeBool,
-		Optional:    true,
-		Default:     false,
-		Description: "Specifies that the view is secure. By design, the Snowflake's `SHOW VIEWS` command does not provide information about secure views (consult [view usage notes](https://docs.snowflake.com/en/sql-reference/sql/create-view#usage-notes)) which is essential to manage/import view with Terraform. Use the role owning the view while managing secure views.",
+		Type:             schema.TypeString,
+		Optional:         true,
+		Default:          BooleanDefault,
+		ValidateDiagFunc: validateBooleanString,
+		DiffSuppressFunc: IgnoreChangeToCurrentSnowflakeValueInShow("is_secure"),
+		Description:      booleanStringFieldDescription("Specifies that the view is secure. By design, the Snowflake's `SHOW VIEWS` command does not provide information about secure views (consult [view usage notes](https://docs.snowflake.com/en/sql-reference/sql/create-view#usage-notes)) which is essential to manage/import view with Terraform. Use the role owning the view while managing secure views."),
 	},
+	"is_temporary": {
```

Comment:
Adjusted the field description and added a test. I think that if Snowflake makes it possible to have a temporary view, we can also support it - it's users' decision to use it or not.

Reply:
That's true, but Terraform is used for infrastructure, not data. I can't think of any use case where this could be used in a meaningful way. Temporary views will only last until the end of the `terraform apply`, and in my opinion, they don't make any sense to have in Terraform because there's no use case for them. It may only lead to confusion when used, because of its lifetime and permadiff in Terraform plans.
cc: @sfc-gh-asawicki

Reply:
As we discussed, we'll keep this field for now.